cmd/promtool/unittest.go

```diff
@@ -190,6 +172,12 @@ func (tg *testGroup) test(mint, maxt time.Time, evalInterval time.Duration, grou
 	}
 	groups := orderedGroups(groupsMap, groupOrderMap)
 
+	// Bounds for evaluating the rules.
+	mint := time.Unix(0, 0).UTC()
+	maxt := mint.Add(tg.maxEvalTime())
+	// Rounding off to nearest Eval time (> maxt).
+	maxt = maxt.Add(evalInterval / 2).Round(evalInterval)
```

Comment:
It's existing code, but this feels wrong.

Why do we need to adjust this at all?

Reply:
I believe it is correct, mostly because the loop below does `ts.Before(maxt); ts = ts.Add(evalInterval)`.

I think it doesn't make sense to go changing this for now, there's a different but related issue of the fact that unit testing rules with different evaluation intervals isn't possible. Which it would be nice to fix, but a little tricky as it may change existing tests (e.g. where rules define an interval but right now the test overrides them).

Reply:
It's more the weirdness of using Round like this, I'd expect a more explicit ceil/floor.

We only need a value larger than maxt, adding a nanosecond would suffice.

> unit testing rules with different evaluation intervals isn't possible.

I'm not too worried about that, rules should always have the same evaluation interval or things could get a bit weird.

Reply:
Ok, I added some tests to convince me this was right and did the easier fix of just checking for equality.